### ISSUE TITLE
module command not found on spack install

### ISSUE THREAD

**Issue by berceanu** [ComputationalRadiationPhysics/picongpu#2770]
Following the [spack install guide](https://picongpu.readthedocs.io/en/0.4.0/install/instructions/spack.html), the command 
```
spack load gcc@7.3.0
```
fails with the above error. That means that `spack compiler add` is never executed.

Even though `spack bootstrap` installs `environment-modules`, they are not added to the runtime path.
The fix is to run
```
source $HOME/src/spack/share/spack/setup-env.sh
```
once again after `spack bootstrap`.

**Comment by ax3l** [comment]
Oh no, how bad!

Then let us change the recommended spack install to
```bash
# get spack
git clone https://github.com/spack/spack.git $HOME/src/spack

# build spack's dependencies via spack :)
./spack/bin/spack bootstrap

# activate the spack environment
source $HOME/src/spack/share/spack/setup-env.sh

# ...
```
?

**Comment by ax3l** [comment]
Sounds good, can do now before turning off the laptop and you can review it tomorrow :)

**Comment by ax3l** [comment]
Merged and again fixed by #2997
